index.js

```diff
@@ -198,39 +134,159 @@ function generateImage(layout, callback) {
 
 
 /**
- * Spritezero can generate 2 kinds of layout objects: {@link DataLayout} and {@link ImgLayout}.
+ * Generate a {Manifest} object with information on where the icons are on the sprite.
+ *
+ * @param  {Layout}   layout    An {@link Layout} Object used to generate the image
+ * @return {Manifest} layout    A Manifest object
+ */
+function generateManifest(layout) {
+    assert(typeof layout === 'object');
+    if (!layout.items.length) return {};
+
+    var obj = {};
+    layout.items.forEach(function(item) {
+        item.names.forEach(function(name) {
+            obj[name] = {
+                width: item.width,
+                height: item.height,
+                x: item.x,
+                y: item.y,
+                pixelRatio: layout.pixelRatio
+            };
+        });
+    });
+    return obj;
+}
+
+function encodeBinaryManifest(manifest) {
+    function writeImage(data, pbf) {
+        pbf.writeStringField(1, data.id);
+        pbf.writeVarintField(2, data.img.width);
+        pbf.writeVarintField(3, data.img.height);
+        pbf.writeVarintField(4, data.img.x);
+        pbf.writeVarintField(5, data.img.y);
+        if (data.img.pixelRatio !== 1) {
+            pbf.writeFloatField(6, data.img.pixelRatio);
+        }
```

Comment:
We could save 3 bytes per glyph here by encoding a varint instead of float if pixelRatio is integer (perhaps with a different tag). Or ban non-integer ratios (which are useless for sprites anyway).

Reply:
I considered that, but ultimately chose fixed floats. Our APIs _do_ support fractional numbers, but we don't currently use them in GL.